### PR TITLE
Import event passwords dialog

### PR DESCRIPTION
**Pull request by nikomakela**
KK-903.

image

### CODE REVIEW

**Comment by nikomakela** [comment]
- [ ] A toast for success and error indicating is needed

**Comment by nikomakela** [comment]
- [ ] Linebreaks are not working in placeholder translation when the translation is fetched from the spreadsheet.

File: src/domain/ticketSystemPassword/ImportTicketSystemPasswordsFormDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    const [passwordsText, setPasswordsText] = React.useState('');
+    const notify = useNotify();
+    const onChangePasswordsText: React.ChangeEventHandler<HTMLTextAreaElement> = (
+      e
+    ) => {
+      setPasswordsText(e.currentTarget.value);
+    };
+
+    const submitPasswords = async () => {
+      // TODO: Handle messaging: errors and succefull import
```
Comment (tuomas777) [inline src/domain/ticketSystemPassword/ImportTicketSystemPasswordsFormDialog.tsx:47]: Is this comment relevant anymore?

File: src/domain/ticketSystemPassword/api/ticketSystemPasswordsApi.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+} from '../../../api/utils/apiUtils';
+import { importEventTicketSystemPasswordsMutation } from '../mutations/TicketSystemPasswordMutations';
+
+export type ErrorType = {
+  field: string;
+  message: string;
+  value: string;
+};
+
+// TODO: Generate
```
Comment (tuomas777) [inline src/domain/ticketSystemPassword/api/ticketSystemPasswordsApi.ts:16]: What about this one? 🙂

File: src/domain/ticketSystemPassword/mutations/TicketSystemPasswordMutations.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+export const importEventTicketSystemPasswordsMutation = gql`
+  mutation ImportTicketSystemPasswordsMutation(
+    $input: ImportTicketSystemPasswordsMutationInput!
+  ) {
+    importTicketSystemPasswords(input: $input) {
+      event {
+        id
+        name
+      }
+      passwords
```
Comment (tuomas777) [inline src/domain/ticketSystemPassword/mutations/TicketSystemPasswordMutations.ts:12]: Are these actually needed?

**Review by tuomas777 — APPROVED** [review]
Added a couple of minor comments, and as discussed some warning seems to emerge in the console when accessing the occurrences page, at least for me.

But in general looks great to me 👍 🐰 🚀

File: src/domain/ticketSystemPassword/ImportTicketSystemPasswordsFormDialog.tsx (hunk 1 above)
Comment (nikomakela) [reply src/domain/ticketSystemPassword/ImportTicketSystemPasswordsFormDialog.tsx:47]: nope

File: src/domain/ticketSystemPassword/api/ticketSystemPasswordsApi.ts (hunk 2 above)
Comment (nikomakela) [reply src/domain/ticketSystemPassword/api/ticketSystemPasswordsApi.ts:16]: actually, it is. I think those should be generated somehow automatically.

File: src/domain/ticketSystemPassword/mutations/TicketSystemPasswordMutations.ts (hunk 3 above)
Comment (nikomakela) [reply src/domain/ticketSystemPassword/mutations/TicketSystemPasswordMutations.ts:12]: This also relates to the "TODO: generate" -comment in the `ticketSystemPasswordsApi.ts`. This model should update the generated schema.

**Pull request merged**